wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -84,9 +86,9 @@ public static synchronized CommandScheduler getInstance() {
 
   // Flag and queues for avoiding ConcurrentModificationException if commands are
   // scheduled/canceled during run
-  private boolean m_inRunLoop;
+  private boolean m_deferScheduleAndCancel;
   private final Set<Command> m_toSchedule = new LinkedHashSet<>();
```

Comment:
This should also be a queue. Sets don't preserve insertion order, which could cause a bug here.